impl/src/main/kotlin/com/android/declarative/internal/DependencyProcessor.kt

```diff
@@ -63,17 +66,62 @@ class DependencyProcessor(
                     addLibraryDependency(dependency)
                 }
                 is ExtensionFunction -> {
-                    throw RuntimeException("Not Supported yet !")
+                    addExtensionDependency(dependency)
                 }
                 is Alias -> {
-                    throw RuntimeException("Version catalogs not supported yet.")
+                    addVersionCatalogDependency(dependency)
+                }
+                is Platform -> {
+                    addPlatformDependency(dependency)
                 }
             }
         }
     }
 
+    private fun addVersionCatalogDependency(dependency: Alias) {
+
+        val versionCatalogIdentifier = dependency.alias.substringBefore('.')
+        val libraryName = dependency.alias.substringAfter('.')
+        val versionCatalog = versionCatalogs.getVersionCatalog(project, versionCatalogIdentifier)
+        val lib = versionCatalog.findLibrary(libraryName)
+        lib.ifPresentOrElse({
+            println("Adding Version catalog ${dependency.alias} to ${dependency.configuration}")
+            dependencyHandler.add(
+                dependency.configuration,
+                it
+            )
+        }) {
+            issueLogger.logger.warning(
+                "$libraryName library not found in version catalog $versionCatalogIdentifier"
+            )
+        }
+    }
+
+    private fun addPlatformDependency(dependency: Platform) {
+        println("Adding platform ${dependency.name} to ${dependency.configuration}")
+        if (dependency.name.contains(".")) {
+            // TODO: Check that the assumption of a notation with . is a version catalog alias.
+            val versionCatalogIdentifier = dependency.name.substringBefore('.')
+            val platformName = dependency.name.substringAfter('.')
```

Comment:
Assuming we have platform dependency only in catalog, correct?

Reply:
no but it comes back to my other comment, I will do a CL to introduce platformRef and platform to disambiguate.